File: src/OrganizationDatabase/OrganizationManagerInterface.java
package OrganizationDatabase;

import exceptions.OrganizationAlreadyPresentedException;
import exceptions.OrganizationNotFoundException;
import lib.ExecutionStatus;

public interface OrganizationManagerInterface {
    String getInfo();

    float getSumOfAnnualTurnover();

    Organization maxByFullName() throws OrganizationNotFoundException;

    void add(Organization organization) throws OrganizationAlreadyPresentedException;

    void add(Organization... newOrganizations) throws OrganizationAlreadyPresentedException;

    ExecutionStatus addIfMax(Organization newOrganization) throws OrganizationAlreadyPresentedException;

    void modifyOrganization(Organization updatedOrganization) throws OrganizationAlreadyPresentedException;

    void removeById(int id) throws OrganizationNotFoundException;

    void removeAllByPostalAddress(Address address);

    Organization removeHead();

    void clear();

    ExecutionStatus saveToFile(String path);

    ExecutionStatus loadFromFile(String path);

    String toYaml();

    String toJson();

    String toCSV();
}

File: src/database/OrganizationManagerInterface.kt
package database

import lib.ExecutionStatus

interface OrganizationManagerInterface {
    val info: String

    val sumOfAnnualTurnover: Float

    suspend fun maxByFullName(): Organization?

    suspend fun add(organization: Organization)

    suspend fun add(vararg newOrganizations: Organization)

    suspend fun addIfMax(newOrganization: Organization): ExecutionStatus

    suspend fun modifyOrganization(updatedOrganization: Organization)

    suspend fun removeById(id: Int) : ExecutionStatus

    suspend fun removeAllByPostalAddress(address: Address)

    suspend fun removeHead(): Organization?

    suspend fun clear()

    suspend fun save(path: String): ExecutionStatus

    suspend fun loadFromFile(path: String): ExecutionStatus

    suspend fun toYaml(): String

    suspend fun toJson(): String

    suspend fun toCSV(): String
}
